File: algorithms/search-algorithms/tasks-js/2-binary-search.js
// The binary search demands that the array of data being sorted in inc way.
// O(log(n))

const binarySearch = (arr, target) => {
  let left = 0;
  let right = target.length - 1;

  while (left < right) {
    const middle = Math.floor((right + left) / 2);
    const valueInTheMiddle = arr[middle];

    if (target === valueInTheMiddle) return valueInTheMiddle;

    if (target < valueInTheMiddle) {
      right = middle;
    } else {
      left = middle;
    }
  }

  return -1;
};

File: algorithms/search-algorithms/python/2-binary-search.py
# The binary search demands that the array of data being sorted in inc way.

def binarySearch(arr, target):
  left = 0
  right = len(list) - 1

  while right >= left:
    middle = (left + right) // 2
    value = arr[middle]

    if value == target:
      return target
    
    if value > target:
      right = middle - 1
    else: 
      left = middle + 1
    
  return -1

list = [10, 20, 30, 40, 50]

print(binarySearch(list, 20))
print(binarySearch(list, 100))